### ISSUE TITLE
Excessive cache clearing through descendantOfTag usage

### ISSUE THREAD

**Issue by hoekema** [neos/neos-ui#3107]
### Description

When editing content somewhere in the pagetree the cache for all parent nodes is cleared. This is caused by the `${Neos.Caching.descendantOfTag(documentNode)}` cache instruction used in `prototype(Neos.Neos:Page) -> head -> javascriptBackendInformation` in the `Neos.Neos.Ui` package.

Offending line: https://github.com/neos/neos-ui/blob/master/Resources/Private/Fusion/Prototypes/Page.fusion#L44

### Steps to Reproduce

1. Install 't3n/neos-debug' and enable the toolbar to check for cache hits/misses
2. Open Neos in a browser, open the website homepage in an incognito/private window
3. Edit content somewhere in the pagetree (below the homepage) in Neos
4. Check if there's a cache hit/miss in the debugger toolbar

#### Expected behavior

The cache on the homepage should not be cleared (depending on installation and custom cache rules of course).

#### Actual behavior

The homepage cache is cleared resulting in cache misses.

### Affected Versions

Probably all versions that include the `Neos.Neos:Page` prototype from the `Neos.Neos:Ui` package.

### Solution

Make the `@cache` `entryTags` conditional based on the fact we're in the backend, e.g.:

```
@cache {
  entryTags {
   2 = ${documentNode.context.inBackend ? Neos.Caching.descendantOfTag(documentNode) : null}
  }
}
```

**Comment by kdambekalns** [comment]
TL;DR: I cannot reproduce this. Can you provide a way to reproduce this with the Neos.Demo site, @hoekema?

---

> The `@cache.entryIdentifier.inBackend = ${documentNode.context.inBackend}` makes sure there are two separate items stored for backend and non-backend modes.
> The `@cache.entryTags` are stored on the cached item without differentiation between backend and non-backend. So from cache clearing perspective both items are cleared when a "descendant" is modified.

But that would mean the same entry tags are added to distinct cache entries – that is hard to believe.

I just set up a demo site with 7.3 and when editing something in the backend, the logs clearly show very limited cache clearing… And the cache entries for the homepage stay untouched according to `t3n/neos-debug` when I edit content on a direct and indirect child document.

Editing a document itself (changing the title of the "Try me" document) of course removes more cache entries, as that influences the navigation.

**Comment by kdambekalns** [comment]
Tried again, I am convinced. Also, I wrote:

> But that would mean the same entry tags are added to distinct cache entries – that is hard to believe.

Of course, the distinct entries have the same tags configured, so they are equal. Nothing suspicious…

I tried your suggested solution, and it indeed solves the issue, AFAICT.
